docs/azure-resources.md

```diff
@@ -0,0 +1,86 @@
+* Test isolation means that one set of tests (say Python) cannot cause another set of tests (say .net) to fail by exhausing things like Hub resources.  For example, if a Python test causes C2D throttling, we don't want this to affect C# tests.
+* A global view of tests means that we can have one dashboard showing all test runs with a single way to find all test results.
+
+Instance and shared resources may be in different Azure subscriptions.
+This is necessary to support developers running under their own subscriptions while still reporting to the global stores.
+
+## Shared THIEF resources
+
+All of these resources are under the subscription `THIEF_SHARED_SUBSCRIPTION_ID` in the `THIEF_SHARED_RESOURCE_GROUP` resource group.
+
+`THIEF_SHARED_KEYVAULT_NAME` has a keyvault with secrets for all shared resources.
+This kevault is used to seed the instance keyvault and many of the secrets in this keyvault are duplicated in the instance keyvault
+
+`THIEF_APP_INSIGHTS_CONNECTION_STRING` is the entrypoint for the shared App Insights intance.
+
+`THIEF_CONTAINER_REGISTRY_HOST` is the shared container registry.
+
+There is also a shared Log Analytics workspace and a shared Azure Storage account, but these are not exposed to test scripts.
+
+## Instance THIEF resources
+
+All of these resources are under the subscription `THIEF_SUBSCRIPTION_ID` in the `THIEF_RESOURCE_GROUP` resource group.
+
+`THIEF_KEYVAULT_NAME` is the keyvault that holds secrets for the thief instance which is being tested.
+
+`THIEF_IOTHUB_NAME` is the Azure IoT Hub instance that is being tested.
+
+`THIEF_DEVICE_PROVISIONING_HOST` is the DPS instance that is being tested.
+
+Container instances for running tests use `THIEF_SUBSCRIPTION_ID` subscription, and go into the `THIEF_RUNS_RESOURCE_GROUP` resource group.
+
+## Fetching secrets
+
+Thief secrets are retrieved by setting `THIEF_SUBSCRIPTION_ID` and `THIEF_KEYVAULT_NAME` environment variables and then calling `scripts/fetch_secrets.sh`
+
+## Creating a new THIEF instance
+
+To create a new thief instance:
+1. Set your environment variables for the shared resources by setting `THIEF_SUBSCRIPTION_ID` and `THIEF_KEYVAULT_NAME` to the shared values and calling `scripts/fetch_secrets.sh`
+```
+(longhaul) bertk@bertk-hp:~/repos/longhaul$ export THIEF_SUBSRIPTION_ID=<REDACTED> && export THIEF_KEYVAULT_NAME=<REDACTED> && source ./scripts/fetch-secrets.sh
+```
+
+2. Use `az account set --subsciption <foo>` to set the subsciption you want to use for your instance resources.  Use `az account list` if you need help finding the subscription id.
+```
+(longhaul) bertk@bertk-hp:~/repos$ az account set --subscription "<REDACTED>"
+```
+
+3. Call `scripts/create-thief-instance.sh`, passing in a resource prefix (such as your email address) to create the thief resources.
+```
+(longhaul) bertk@bertk-hp:~/repos/longhaul$ ./scripts/create-thief-instance.sh bertk7 --location westus2
```

Comment:
I'm thinking this won't work for me on Windows. Thoughts on cross-platform support?

Reply:
There's 2 issues here.  `create-thief-resources.sh` is one, but it only gets called once if you need to run against a private hub.  The other issue is `fetch-secrets.sh` which you use to get secrets from the vault and into the environment.  We definitely want a Windows version of fetch-secrets.sh.  I'm not sure about create-thief-instance.

Reply:
Are you saying I need to run Linux to participate in Thief?

Reply:
Right now, yes, Linux is required.  Sort of.    

I would say that you need to run Linux to use some of the more automated aspects.  

I agree that this is a problem that we need to solve.  Bicep is a good start, but there are still things it doesn't do.  We still need scripts for:
1. creating DPS groups and enrollments.
2. fetching secrets (currently into environment variables)
3. building and launching containers.

We can make all of these things work cross-platform.   We just have to decide how and what priority.

(Side note: Windows container support is getting better, but still lags behind Linux.  We may need a different environment (like VMs) when we want automated tests under Windows).

Reply:
I thought PowerShell was a cross platform option. Would that work for both Linux and Windows?

Reply:
To be clear, I'm talking about the developer environment, which is where I assume this script is being run.

Reply:
Yes, the developer environment.  PowerShell is an option.  There's also bash on Windows, which I assume goes through WSL.

Reply:
WSL is "sorta" cross platform, but not as smooth an experience as PowerShell (I'd assume, but not having used PowerShell on Linux I can't say it is great).

I'll accept WSL for now, but perhaps we can look at streamlining this experience.